docs/spec.md

```diff
@@ -2679,6 +2679,46 @@ op, but the result shape is specified dynamically via `output_shape`.
 
 &nbsp;[More Examples](https://github.com/openxla/stablehlo/tree/main/stablehlo/tests/interpret/dynamic_iota.mlir)
 
+### dynamic_reshape
+
+#### Semantics
+
+This operation is functionally identical to
+[reshape](https://github.com/openxla/stablehlo/blob/main/docs/spec.md#reshape)
+op, except the result shape is specified dynamically via `output_shape`.
+
+#### Inputs
+
+| Label | Name           | Type                                         | Constraints |
+|-------|----------------|----------------------------------------------|-------------|
+| (I1)  | `operand`      | tensor or quantized tensor                   | (C1), (C2)  |
+| (I2)  | `output_shape` | 1-dimensional tensor constant of type `si64` | (C2)        |
+
+#### Outputs
+
+| Name     | Type                       | Constraints |
+|----------|----------------------------|-------------|
+| `result` | tensor or quantized tensor | (C1), (C2)  |
+
+#### Constraints
+
+* (C1) `element_type(result)` is given by:
+  * `element_type(operand)`, if `!is_per_axis_quantized(operand)`.
+  * `element_type(operand)` except that `quantization_dimension(operand)` and
+    `quantization_dimension(result)` may differ, otherwise.
+* (C2) `size(output_shape) = rank(result)`.
```

Comment:
Please add the constraints C2, and C3 from reshape op back and add C4. `size(output_shape) = rank(result)`

Reply:
Added C3 back to the spec to match with reshape and applicable for dynamic_reshape.

dynamic_reshape op specific constraint added as C4